hooks/tk-multi-publish2/basic/publish_file.py

```diff
@@ -0,0 +1,341 @@
+
+        # arguments for publish registration
+        self.logger.info("Registering publish...")
+        publish_data= {
+            "tk": publisher.sgtk,
+            "context": item.context,
+            "comment": item.description,
+            "path": path,
+            "name": publish_name,
+            "version_number": version_number,
+            "thumbnail_path": item.get_thumbnail_as_path(),
+            "published_file_type": publish_type,
+            "dependency_paths": dependency_paths
+        }
+
+        # log the publish data for debugging
+        self.logger.debug(
+            "Populated Publish data...",
+            extra={
+                "action_show_more_info": {
+                    "label": "Publish Data",
+                    "tooltip": "Show the complete Publish data dictionary",
+                    "text": "<pre>%s</pre>" % (pprint.pformat(publish_data),)
+                }
+            }
+        )
+
+        # create the publish and stash it in the item properties for other
+        # plugins to use.
+        item.properties["sg_publish_data"] = sgtk.util.register_publish(
+            **publish_data)
+        self.logger.info("Publish registered!")
+
+    def finalize(self, settings, item):
+        """
+        Execute the finalization pass. This pass executes once
+        all the publish tasks have completed, and can for example
+        be used to version up files.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        """
+
+        publisher = self.parent
+
+        # get the data for the publish that was just created in SG
+        publish_data = item.properties["sg_publish_data"]
+
+        # ensure conflicting publishes have their status cleared
+        publisher.util.clear_status_for_conflicting_publishes(
+            item.context, publish_data)
+
+        self.logger.info(
+            "Cleared the status of all previous, conflicting publishes")
+
+        path = item.properties["path"]
+        self.logger.info(
+            "Publish created for file: %s" % (path,),
+            extra={
+                "action_show_in_shotgun": {
+                    "label": "Show Publish",
+                    "tooltip": "Open the Publish in Shotgun.",
+                    "entity": publish_data
+                }
+            }
+        )
+
+    def _get_publish_type(self, extension, settings):
+        """
+        Get a publish type for the supplied extension and publish settings.
+
+        :param extension: The file extension to find a publish type for
+        :param settings: The publish settings defining the publish types
+
+        :return: A publish type or None if one could not be found.
+        """
+
+        # ensure lowercase and no dot
+        if extension:
+            extension = extension.lstrip(".").lower()
```

Comment:
What if there is a file name with multiple . characters? Why not use os.path.splitext?